### PR TITLE
Towards BG-1080: Replace DafDirect with Chariot

### PR DESCRIPTION
**Pull request by misicnenad**
## Explanation of the solution
**The backend (lambda) is not working as we're missing the necessary API credentials. The webapp side of the implementation works right up to the moment the request is sent to the aforementioned lambda. Other than that, the code is ready for review.**

Related lambda PR https://github.com/AngelProtocolFinance/devops/pull/480

**Note for @SovereignAndrey - we'll need to set `PUBLIC_CHARIOT_CONNECT_ID` in our build flows (see how to get it [in official docs](https://givechariot.readme.io/reference/integrating-connect#add-chariot-connect)).**

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
- go to http://localhost:4200/donate/53
- select "DAF" tab

## UI changes to review

image

image

image

### CODE REVIEW

File: src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            />
+          }
+        />
+        <p className="text-sm text-gray-d2 dark:text-gray mt-4">
+          Please click the button below and follow the instructions provided to
+          complete your donation
+        </p>
+
+        <button
+          disabled={methods.formState.isSubmitting}
```
Comment (ap-justin) [inline src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx:130]: found in [docs](https://react-hook-form.com/docs/useform/formstate) that formState is best destructured
image

File: src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          className="btn-orange btn-donate mt-2"
+          type="submit"
+        >
+          <LoadText
```
Comment (ap-justin) [inline src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx:134]: might not need load text, as submission is just synchronous

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx (hunk 1 above)
Comment (misicnenad) [reply src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx]: Interesting :thinking: I understand why they recommend that, but it's still unintuitive.

Fixed.

File: src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx (hunk 2 above)
Comment (misicnenad) [reply src/components/donation/Steps/DonateMethods/ChariotConnect/Form.tsx]: Removed `disabled` prop assignments from all */DonateMethods/\*/Form* files as they have basically the same logic.

**Review by ap-justin — APPROVED** [review]

**Pull request merged**